core/src/main/java/hudson/model/Computer.java

```diff
@@ -942,18 +942,18 @@ public String getSearchUrl() {
     }
 
     /**
-     * Gets the environment variables of the JVM on this computer.
+     * Returns cached environment variables (copy to prevent modification) for the JVM on this computer.
      * If this is the master, it returns the system property of the master computer.
      */
     public EnvVars getEnvironment() throws IOException, InterruptedException {
         EnvVars cachedEnvironment = this.cachedEnvironment;
         if (cachedEnvironment != null) {
```

Comment:
I don't think it's a big deal if there are two consecutive calls to this method and EnvVars.getRemote(getChannel()) is invoked multiple times, cachedEnvironment will eventually be set.